pkg/controllers/gatewaydns/endpoint_slice_reconciler.go

```diff
@@ -122,57 +122,67 @@ type ClusterDiff struct {
 	changed   []discoveryv1beta1.EndpointSlice
 }
 
-func (e *EndpointSliceReconciler) diffCluster(ctx context.Context, gatewayDNSNamespacedName types.NamespacedName, clusterClient client.Client, desiredEndpointSlices []discoveryv1beta1.EndpointSlice) (ClusterDiff, error) {
+func (e *EndpointSliceReconciler) diffCluster(ctx context.Context,
+	log logr.Logger,
+	gatewayDNSNamespacedName types.NamespacedName,
+	clusterClient client.Client,
+	desiredClusterGateways []ClusterGateway) (ClusterDiff, error) {
+
 	existingEndpointSliceList := &discoveryv1beta1.EndpointSliceList{}
 	err := clusterClient.List(ctx, existingEndpointSliceList, client.InNamespace(e.Namespace))
 	if err != nil {
 		return ClusterDiff{}, err
 	}
 
-	existingEndpointSliceMap := make(map[string]discoveryv1beta1.EndpointSlice)
-	for _, item := range existingEndpointSliceList.Items {
-		if _, ok := item.Annotations[connectivityv1alpha1.DNSHostnameAnnotation]; !ok {
+	existingEndpointSliceMap := make(map[string]discoveryv1beta1.EndpointSlice, len(existingEndpointSliceList.Items))
+	for _, existingEndpointSlice := range existingEndpointSliceList.Items {
+		if _, ok := existingEndpointSlice.Annotations[connectivityv1alpha1.DNSHostnameAnnotation]; !ok {
 			continue
 		}
 
-		existingGatewayDNSNamespacedName, ok := item.Annotations[connectivityv1alpha1.GatewayDNSRefAnnotation]
+		existingGatewayDNSNamespacedName, ok := existingEndpointSlice.Annotations[connectivityv1alpha1.GatewayDNSRefAnnotation]
 		if !ok || existingGatewayDNSNamespacedName != gatewayDNSNamespacedName.String() {
 			continue
 		}
 
-		existingEndpointSliceMap[endpointSliceKey(item)] = item
+		existingEndpointSliceMap[EndpointSliceKey(existingEndpointSlice)] = existingEndpointSlice
 	}
```

Comment:
consider extracting to a private method, maybe named `getExistingEndpointSlices`??